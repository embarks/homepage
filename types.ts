export enum FORTUNE_TYPES {
  joke = 'jokes',
  quote = 'quotes',
  poem = 'poems'
}

export type Fortune = {
  type: FORTUNE_TYPES
  lines: string[]
  author?: string
  title?: string
}
