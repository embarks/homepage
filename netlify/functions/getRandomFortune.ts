import { Fortune, FORTUNE_TYPES } from '../../types'

export async function getRandomFortune(): Promise<Fortune> {
  const fortuneTypes = Object.values(FORTUNE_TYPES)
  const randomType = Math.floor(Math.random() * fortuneTypes.length)
  const fortuneType = fortuneTypes[randomType]
  try {
    const { fortunes } = await import(`./fortunes/${fortuneType}.json`)
    const randomFortuneIndex = Math.floor(Math.random() * fortunes.length)
    const fortune = fortunes.fortunes[randomFortuneIndex]
    return fortune
  } catch (error) {
    return {
      type: FORTUNE_TYPES.joke,
      lines: ['The world will end in 5 minutes.  Please log out.']
    }
  }
}
