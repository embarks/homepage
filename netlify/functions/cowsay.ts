import { Handler } from '@netlify/functions'
import cowsay from 'cowsayjs'
import { Fortune } from '../../types'
import { getRandomFortune } from './getRandomFortune'

const handler: Handler = async (event, context) => {
  const { lines }: Fortune = await getRandomFortune()

  return {
    headers: {
      'Access-Control-Allow-Origin': 'http://localhost:8080'
    },
    statusCode: 200,
    body: JSON.stringify({
      message: cowsay.moo(lines.join('\n'))
    })
  }
}

export { handler }
