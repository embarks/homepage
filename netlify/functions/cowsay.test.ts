// test cowsay netlify function
import { FORTUNE_TYPES } from '../../types'
import { handler } from './cowsay'
import * as utils from './getRandomFortune'

describe('cowsay', () => {
  it('should return a cow saying something', async () => {
    expect(async () => {
      const response = await handler({} as any, {} as any)
      if (response) {
        const { body } = response
        const { message } = JSON.parse(body ?? "{ message: '' }")
        console.log(message)
      }
    }).not.toThrowError()

    const getFortune = jest.spyOn(utils, 'getRandomFortune')
    getFortune.mockResolvedValue({
      type: FORTUNE_TYPES.quote,
      lines: ['Hello World']
    })
    expect(handler({} as any, {} as any)).resolves.toMatchSnapshot()
  })
})
