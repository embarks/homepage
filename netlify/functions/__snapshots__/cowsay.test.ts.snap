// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cowsay should return a cow saying something 1`] = `
Object {
  "body": "{\\"message\\":\\" _____________ \\\\n< Hello World >\\\\n ------------- \\\\n        \\\\\\\\   ^__^\\\\n         \\\\\\\\  (oo)\\\\\\\\_______\\\\n            (__)\\\\\\\\       )\\\\\\\\/\\\\\\\\\\\\n                ||----w |\\\\n                ||     ||\\"}",
  "headers": Object {
    "Access-Control-Allow-Origin": "http://localhost:8080",
  },
  "statusCode": 200,
}
`;
